lib/Gateways/Version1/Gateway.php

```diff
@@ -265,11 +265,12 @@ public function generateMerchantLink($transaction)
         return $transaction->object();
     }
 
-    public function generateHostedPageAccessToken($transaction)
+    public function generateHostedPageAccessToken($transaction, $readOnly = 0)
```

Comment:
Looking through this file, it seems this method would now be the only method accepting more than one parameter.
Following the existing pattern, we'd have to find a way to integrate `readOnly` as a member with a getter & setter within $transaction (which is of mixed type).

Typically the `$transaction` parameter is an object with enough data to perform whatever the given action is. Are you sure we want to start breaking this pattern?